internal/buildengine/engine.go

```diff
@@ -1308,3 +1294,79 @@ func (e *Engine) watchForPluginEvents(originalCtx context.Context) {
 		}
 	}
 }
+
+// processDeploymentQueue handles the deployment queue and groups pending deployments into changesets
+func (e *Engine) processDeploymentQueue(ctx context.Context) {
+	// Subscribe to schema changes
+	schemaChanges := make(chan schemaeventsource.Event, 128)
+	e.schemaChanges.Subscribe(schemaChanges)
+	defer e.schemaChanges.Unsubscribe(schemaChanges)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+
+		case modules := <-e.deploymentQueue:
+			// If we have an active changeset, add to pending deployments
+			if _, ok := e.activeChangesetKey.Get(); ok {
+				e.pendingDeployments = append(e.pendingDeployments, modules...)
```

Comment:
What triggers the pending deployments to Deploy if no further events hit the deploymentQueue?

Reply:
Actually never mind, I see they get copied back into the queue when the changeset ends.